### ISSUE TITLE
pyml2ds

### ISSUE THREAD

**Issue by jk5153** [sassoftware/python-sasctl#120]
I have a xgboost model built in Python outside of SAS Viya.  I’m trying to convert the Python xgboost model to SAS scoring code.  According to the website:  https://sassoftware.github.io/python-sasctl/api/sasctl.utils.pyml2ds.html, there is a method “pyml2ds” in the sasctl package which should be able to do this in Python.  It does say this method is experimental and may be modified or removed without warning.  Has the method "pyml2ds" been removed from the sasctl package?

I'm getting a name error when trying this method (see below).

![2022-04-13_8-48-01](https://user-images.githubusercontent.com/85761488/163263942-54b00eff-9be6-46e9-8bd7-056e2dce2ecc.png)

**Comment by smlindauer** [comment]
In order to reference the pyml2ds function, you will need to import from the utils folder:

`from sasctl.utils import pyml2ds`

As a general warning, this function is over three years old and will likely not conform with the DS2 format expected by SAS Viya. Furthermore, if you are using Viya 4, there is no need to convert the python score code to DS2.

A more recent version function follows the [pzmm example notebook](https://github.com/sassoftware/python-sasctl/blob/master/examples/pzmmModelImportExample.ipynb), which will take in our model pickle and create python score code and upload the model to SAS Model Manager. If the function detects Viya 3.5, then it will produce DS2 code to also be imported into the model.

**Comment by jk5153** [comment]
Thanks for the info.  I was able to reference the pyml2ds function using the following:  from sasctl.utils.pyml2ds import pyml2ds.  The pyml2ds function still produces valid SAS data step code.  The SAS data step code produced xgboost model score matched the python produced xgboost model score to the 4th decimal place for 99.95% of the records.  The model pickle file has slightly different values for the split conditions than what is outputted for the SAS data step model code and the dump file.  The split conditions for the the SAS data step model code and the dump file are more precise than the model pickle file.  I'm not sure why this is happening.

I did try the pzmm example notebook you provided and was able to load the model into SAS Model Manager.  However, there are some configuration issues that won't allow me to score the registered model.  Once this is fixed, I'm hoping this score will match exactly the python produced xgboost model score for all records.
